editor/scene_tree_dock.cpp

```diff
@@ -1919,7 +1919,9 @@ void SceneTreeDock::_do_reparent(Node *p_new_parent, int p_position_in_parent, V
 		// `move_child` + `get_index` doesn't really work for internal nodes.
 		ERR_FAIL_COND_MSG(p_nodes[ni]->get_internal_mode() != INTERNAL_MODE_DISABLED, "Trying to move internal node, this is not supported.");
 
-		if (p_nodes[ni]->get_parent() != p_new_parent || p_position_in_parent + ni != p_nodes[ni]->get_index(false)) {
+		const bool same_parent = p_nodes[ni]->get_parent() == p_new_parent;
+		const int new_index = p_position_in_parent > -1 ? p_position_in_parent + ni : p_new_parent->get_child_count(false) + ni;
+		if (!same_parent || new_index != p_nodes[ni]->get_index(false) && (new_index - 1 < 0 || new_index - 1 >= p_new_parent->get_child_count(false) || p_new_parent->get_child(new_index - 1, false) != p_nodes[ni])) {
```

Comment:
Parentheses required to clear up ambiguity of grouping here, is it:
```suggestion
		if (!same_parent || new_index != p_nodes[ni]->get_index(false) && (new_index - 1 < 0 || (new_index - 1 >= p_new_parent->get_child_count(false) || p_new_parent->get_child(new_index - 1, false) != p_nodes[ni]))) {
```
Or
```suggestion
		if ((!same_parent || new_index != p_nodes[ni]->get_index(false)) && (new_index - 1 < 0 || new_index - 1 >= p_new_parent->get_child_count(false) || p_new_parent->get_child(new_index - 1, false) != p_nodes[ni])) {
```